src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -255,10 +257,19 @@ export function getInputType(
 
     function getInputModelType(m: Model): InputType {
         const intrinsicName = getIntrinsicModelName(program, m);
-        if (intrinsicName && intrinsicName === "string") {
-            const values = getKnownValues(program, m);
-            if (values) {
-                return getInputModelForExtensibleEnum(m, values);
+        if (intrinsicName) {
+            switch (intrinsicName) {
+                case "string":
+                    const values = getKnownValues(program, m);
+                    if (values) {
+                        return getInputModelForExtensibleEnum(m, values);
+                    };
+                default:
+                    return {
+                        Name: m.name,
```

Comment:
I don't use `mapCadlIntrinsicModelToCsharpModel` here, because it will erase the original model's name. For example, `ResourceLocation` will be changed into `string`.
The cost is that in many existing `cadl.json` there will be changes of property names. But that won't change the generated codes.

Reply:
If we wrap an model to primitive model, the model name should change too. The origin model is not emitted anymore.

Reply:
The primitive model is already represented by the `Kind`. I suggest we should keep the name for debug purpose.